cmake/FindLibUV.cmake

```diff
@@ -31,11 +31,7 @@ if(LIBUV_USE_STATIC)
     "${CMAKE_STATIC_LIBRARY_PREFIX}uv${CMAKE_STATIC_LIBRARY_SUFFIX}")
 endif(LIBUV_USE_STATIC)
 
-if(MSVC)
-  list(APPEND LIBUV_NAMES libuv)
-else()
-  list(APPEND LIBUV_NAMES uv)
-endif()
+list(APPEND LIBUV_NAMES uv)
```

Comment:
The build logs show that luv is being linked to the MSYS2 libuv package instead of the one built by the dependencies. But it doesn't appear to be caused by the changes in this PR because the logs from master show the same thing.
```
Found LIBUV: C:/msys64/mingw64/lib/libuv.dll.a;iphlpapi;psapi;userenv;ws2_32  
```

Reply:
@justinmk I am doing `add_library (uv ..)` [here](https://github.com/neovim/neovim/pull/8434/files#diff-ca30aab4c551663b547e390f3d169775R7). `CMAKE_SHARED_LIBRARY_PREFIX` is empty in MSVC, so I think that this is no longer necessary. I think this setting is due to that we was using MinGW libuv on MSVC, because at that time MSVC could not build libuv.